src/analysis/processing/qgsalgorithmextractlayoutmapextent.cpp

```diff
@@ -0,0 +1,127 @@
+/***************************************************************************
+                         qgsalgorithmextractlayoutmapextent.cpp
+                         ---------------------
+    begin                : March 2019
+    copyright            : (C) 2019 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmextractlayoutmapextent.h"
+#include "layout/qgslayoutitemregistry.h"
+#include "layout/qgslayoutitemmap.h"
+#include "layout/qgslayout.h"
+#include "qgsprocessingoutputs.h"
+///@cond PRIVATE
+
+QString QgsLayoutMapExtentToLayerAlgorithm::name() const
+{
+  return QStringLiteral( "printlayoutmapextenttolayer" );
+}
+
+QString QgsLayoutMapExtentToLayerAlgorithm::displayName() const
+{
+  return QObject::tr( "Print layout map extent to layer" );
+}
+
+QStringList QgsLayoutMapExtentToLayerAlgorithm::tags() const
+{
+  return QObject::tr( "layout,composer,composition,visible" ).split( ',' );
+}
+
+QString QgsLayoutMapExtentToLayerAlgorithm::group() const
+{
+  return QObject::tr( "Cartography" );
+}
+
+QString QgsLayoutMapExtentToLayerAlgorithm::groupId() const
+{
+  return QStringLiteral( "cartography" );
+}
+
+QString QgsLayoutMapExtentToLayerAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Creates a polygon layer containing the extent of a print layout map item." );
+}
+
+void QgsLayoutMapExtentToLayerAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterLayout( QStringLiteral( "LAYOUT" ), QObject::tr( "Print layout" ) ) );
+  addParameter( new QgsProcessingParameterLayoutItem( QStringLiteral( "MAP" ), QObject::tr( "Map item" ), QVariant(), QStringLiteral( "LAYOUT" ), QgsLayoutItemRegistry::LayoutMap ) );
```

Comment:
Ok -- this would require the numeric outputs to be nulled if > 1 map is found, but I think that's ok. I don't want to drop those numeric outputs as they are useful for models.